File: app/src/main/java/me/kirimin/mitsumine/network/api/EntryInfoApi.java
package me.kirimin.mitsumine.network.api;

import com.android.volley.RequestQueue;

import org.json.JSONObject;

import rx.Observable;

public class EntryInfoApi {

    private static final String REQUEST_URL = "http://b.hatena.ne.jp/entry/jsonlite/?url=";

    public static Observable<JSONObject> request(final RequestQueue requestQueue, final String url) {
        return ApiAccessor.request(requestQueue, REQUEST_URL + url);
    }
}

File: app/src/main/java/me/kirimin/mitsumine/network/api/EntryInfoApi.kt
package me.kirimin.mitsumine.network.api

import android.content.Context
import me.kirimin.mitsumine.model.EntryInfo
import me.kirimin.mitsumine.network.api.parser.EntryInfoApiParser

import rx.Observable

public class EntryInfoApi {
    companion object {

        private val REQUEST_URL = "http://b.hatena.ne.jp/entry/jsonlite/?url="

        public fun request(context: Context, url: String): Observable<EntryInfo> {
            return ApiAccessor.request(context, REQUEST_URL + url)
                    .map { response -> EntryInfoApiParser.parseResponse(response) }
        }
    }
}